spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/propagation/TagPropagationFinishedSpanHandler.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.propagation;
+
+import brave.handler.FinishedSpanHandler;
+import brave.handler.MutableSpan;
+import brave.propagation.ExtraFieldPropagation;
+import brave.propagation.TraceContext;
+import org.springframework.cloud.sleuth.autoconfig.SleuthProperties;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.Objects.nonNull;
+
+/**
+ * Finish span handler which adds extra propagation fields to span tags, so spans could be
+ * looked up by it's baggage.
+ *
+ * @author Taras Danylchuk
+ * @since 2.1.0
+ */
+public class TagPropagationFinishedSpanHandler extends FinishedSpanHandler {
+
+	private final SleuthProperties sleuthProperties;
+
+	private final SleuthTagPropagationProperties tagPropagationProperties;
+
+	public TagPropagationFinishedSpanHandler(SleuthProperties sleuthProperties,
+			SleuthTagPropagationProperties tagPropagationProperties) {
+		this.sleuthProperties = sleuthProperties;
+		this.tagPropagationProperties = tagPropagationProperties;
+	}
+
+	@Override
+	public boolean handle(TraceContext context, MutableSpan span) {
+		if (context == null) {
+			return true;
+		}
```

Comment:
If you update to Brave `5.6.0` then you can add another exit condition of: `!context.isLocalRoot()` that way this doesn't tag every span and blow up your storage requirements.

PS: I have a feeling someone else will block this being merged until that is done

Reply:
I don't really aware about this condition, so if members will require it, I will add - no problem